### PR TITLE
[PW-2022] changes to get it working

### PR DESCRIPTION
**Pull request by rikterbeek**
**Description**

Changes to get PW-2022 working
**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

File: Model/AdyenOrderPaymentStatus.php [hunk 1]
```diff
@@ -55,37 +55,30 @@ public function __construct(
...
+        ) {
+            $additionalInformation = $payment->getAdditionalInformation();
+            return $this->adyenHelper->buildThreeDS2ProcessResponseJson(
+                $additionalInformation['threeDSType'],
+                $additionalInformation['threeDS2Token']
+            );
         }
-
-        return $result;
+        return true;
```
Comment (cyattilakiss) [inline Model/AdyenOrderPaymentStatus.php]: Do we handle the boolean true return?

File: view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js [hunk 2]
```diff
@@ -339,7 +357,10 @@ define(
                         ).done(
                         function (response) {
                             self.afterPlaceOrder();
-                            adyenPaymentService.getOrderPaymentStatus(response, self.validateThreeDS2OrPlaceOrder.bind(self));
+                            adyenPaymentService.getOrderPaymentStatus(response)
+                                .done(function (responseJSON) {
```
Comment (cyattilakiss) [inline view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js]: this way is nicer!

File: view/frontend/web/js/view/payment/method-renderer/adyen-oneclick-method.js [hunk 3]
```diff
@@ -184,6 +207,7 @@ define(
                          * @returns {boolean}
                          */
                         placeOrder: function (data, event) {
+                            debugger;
```
Comment (cyattilakiss) [inline view/frontend/web/js/view/payment/method-renderer/adyen-oneclick-method.js]: Is it WIP?

Comment (rikterbeek) [reply view/frontend/web/js/view/payment/method-renderer/adyen-oneclick-method.js]: it was but it is working so now removed it.

File: Model/AdyenOrderPaymentStatus.php (hunk 1 above)
Comment (rikterbeek) [reply Model/AdyenOrderPaymentStatus.php]: Yes it will continue to the success page.

File: Model/AdyenOrderPaymentStatus.php [hunk 4]
```diff
@@ -55,37 +55,30 @@ public function __construct(
...
-            throw new \Magento\Framework\Exception\LocalizedException(__('This order no longer exists.'));
+        if ($payment->getMethod() === AdyenCcConfigProvider::CODE ||
+            $payment->getMethod() === AdyenOneclickConfigProvider::CODE
+        ) {
+            $additionalInformation = $payment->getAdditionalInformation();
+            return $this->adyenHelper->buildThreeDS2ProcessResponseJson(
+                $additionalInformation['threeDSType'],
+                $additionalInformation['threeDS2Token']
+            );
         }
```
Comment (acampos1916) [inline Model/AdyenOrderPaymentStatus.php]: $this->orderRepository->get($orderId); can still throw an exception if the orderId supplied does not exist. Shouldn't we keep the try/catch block to handle it?

Comment (rikterbeek) [reply Model/AdyenOrderPaymentStatus.php]: @acampos1916 I can indeed but the front-end is not able to catch this exception as this time and as well quite a lot of work to implement as it should never happen. I believe this should give a server error so the merchant during development will notice that they probably build a plugin that changed the response causing this to fail. I am open for any other suggestion but believe most of them will need quite a lot of development for it to make sure the front-end will understand this error.

**Pull request merged**